shared/src/ledger/eth_bridge/vp/mod.rs

```diff
@@ -67,11 +119,18 @@ where
             verifiers_len = verifiers.len(),
             "Ethereum Bridge VP triggered",
         );
+
+        // first check if Nam is being escrowed
+        if keys_changed.contains(&balance_key(&xan(), &super::ADDRESS)) {
+            return Ok(self.check_escrow(verifiers));
```

Comment:
Then escrowing will never pass since it doesn't perform a transfer between erc20s. The `extract_valid_keys_changed` will fail since the changed keys will be empty.  A tx should either escrow nam or transfer erc20s, not both at the same time imo.

Reply:
I've started adding tests in `bat/ethbridge/eth-bridge-vp-escrow-nam`

Reply:
Yeah it should be one or the other, escrowing nam or transferring ERC20s. But this VP as it is now looks like it would accept a transaction that sent some NAM to the Eth bridge, regardless of what else the transaction is changing. We should have something like the below:

```rust

    // we aren't concerned with keys that changed outside of our account
    let keys_changed: HashSet<_> = keys_changed
        .iter()
        .filter(|key| storage::is_eth_bridge_key(key))
        .collect();
```

moved out of `extract_valid_keys_changed` into `validate_tx`. We could dispatch the correct VP logic depending on the contents of `keys_changed`

Reply:
I thought about that earlier, so yeah, I can do that.